main.go

```diff
@@ -493,9 +494,69 @@ func clean(state *state.HelmState, errs []error) error {
 
 func getArgs(c *cli.Context, state *state.HelmState) []string {
 	args := c.String("args")
+	argsMap := map[string]string{}
+
 	if len(args) > 0 {
-		state.HelmDefaults.Args = strings.Split(args, " ")
+		argsVals := strings.Split(args, " ")
+		for _, arg := range argsVals {
+			argVal := strings.SplitN(arg, "=", 2)
+			if len(argVal) > 1 {
+				arg := argVal[0]
+				value := argVal[1]
+				argsMap[arg] = value
+			} else {
+				arg := argVal[0]
+				argsMap[arg] = ""
+			}
+		}
 	}
+	if len(state.HelmDefaults.Args) > 0 {
+		for _, arg := range state.HelmDefaults.Args {
+			argVal := strings.SplitN(arg, "=", 2)
+			arg := argVal[0]
+			if _, exists := argsMap[arg]; !exists {
+				if len(argVal) > 1 {
+					argsMap[arg] = argVal[1]
+				} else {
+					argsMap[arg] = ""
+				}
+			}
+		}
+	}
+
+	setDedicatedKeys(state, argsMap)
+
+	var argArr []string
+
+	for key, val := range argsMap {
+		if val != "" {
+			argArr = append(argArr, fmt.Sprintf("%s=%s", key, val))
+		} else {
+			argArr = append(argArr, fmt.Sprintf("%s", key))
+		}
+	}
+
+	state.HelmDefaults.Args = argArr
 
 	return state.HelmDefaults.Args
 }
+
+func setDedicatedKeys(state *state.HelmState, argsMap map[string]string) {
+
+	helmdefaults := reflect.Indirect(reflect.ValueOf(state.HelmDefaults))
```

Comment:
Nice try! But would you mind if I suggest something like a chain of verbose code like:

```go
if _, exists := argsMap["--tiller-namespace"]; ok && state.HelmDefaults.TillerNamespace != "" {
  argsMap["--tiller-namspace"] = state.HelmDefaults.TillerNamespace
}
```

Or wrap it into a func like:

```golang
setDefaultValue(argsMap, "--tiller-namespace", state.HelmDefaults.TillerNamespace)
setDefaultValue(argsMap, "--kube-context", state.HelmDefaults.KubeContext)
```

The reflection way is definitely DRY, but it seems to require too much engineering effort to maintain compared to the verbose alternatives above, especially when we are to support `--tiller-connection-timeout int` which results in an another field tag (=`arg-type`?) or an another `if`.

Reply:
Sure ill switch from reflection to more verbose checks. Thanks for looking into pull request. will make updates